kv/collection.go

```diff
@@ -403,3 +403,17 @@ func (c Collection[K, V]) UnlessNotEmpty(
 ) Collection[K, V] {
 	return c.WhenEmpty(f)
 }
+
+func (c Collection[K, V]) Forget(k K) (Collection[K, V], error) {
```

Comment:
Wouldn't it be better not to return a new collection, but to remove the data from the existing one?

Reply:
I don't think so, here is the thing. 

Mutable state instead of returning a copy can be a problem when we are discussing about concurrency. Pointers are not thread safe.

Consistency design of all methods already implemented in Go collections 😄

Reply:
I agree with you, but have you ever considered a scenario where I need to remove a set of keys?